index.js

```diff
@@ -54,6 +59,17 @@ module.exports = function * (next) {
     }
   }
 
+  //Adjust infinite end
+  if (end == Number.POSITIVE_INFINITY) {
```

Comment:
1. Sure
2. Infinity can be reassigned but if you prefer that style I can change it.